ui/src/services/utils.js

```diff
@@ -51,6 +51,81 @@ export const sortSelector = createSelector(
   list => list
 );
 
+/**
+ * This function will sort the `array` by capacity.
+ * The capacity should follow k8s regex rules.
+ * Each elements is an object that have a storageCapacity field.
+ *
+ * @param {array} list - The array that will be sort
+ * @param {string} sortBy - The field that will be sort
+ * @param {string} sortDirection - The direction of the sort.
+ *
+ * @example
+ * const capacities = [
+ *  { capacity: '1Ki' },
+ *  { capacity: '1Gi' },
+ *  { capacity: '100Mi' },
+ * ];
+ *
+ * const sortedCapacity = sortCapacity(capacities, 'capacity', 'DESC')
+ */
+export const sortCapacity = createSelector(
+  (list = [], sortBy = 'storageCapacity', sortDirection = 'ASC') => {
+    if (
+      Array.isArray(list) &&
+      typeof sortBy === 'string' &&
+      typeof sortDirection === 'string'
+    ) {
+      const k8sCapacityRegex = /^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$/;
+      const notSortableList = list.filter(
+        item => !k8sCapacityRegex.test(item?.[sortBy])
+      );
+
+      const sortedList = list
+        .filter(item => k8sCapacityRegex.test(item?.[sortBy]))
+        .map(item => {
+          const unit = item[sortBy].slice(-2);
```

Comment:
I think the `size` and `unit` could be extracted with the `k8sCapacityRegex` regex which would be simpler (with a little twist of the regex maybe).
I'll do some tests on my side and come back to you if I've get something good :slightly_smiling_face: